File: row-change-event-channel/src/test/scala/nl/haploid/event/channel/repository/RowChangeEventRepositoryIT.scala
package nl.haploid.event.channel.repository

import nl.haploid.event.channel.{AbstractIT, SlowTest}
import org.springframework.beans.factory.annotation.Autowired

class RowChangeEventRepositoryIT extends AbstractIT {

  @Autowired val repository: RowChangeEventRepository = null

  "Row change event repository" should "be injected" taggedAs SlowTest in {
    setup
    repository should not be null
  }

  "Row change event repository" should "find all events" taggedAs SlowTest in {
    setup
    val event = new RowChangeEvent
    event.tableName = "bob"
    event.rowId = 123
    repository.save(event)
    repository.findAll should have length 1
  }
}

File: row-change-event-channel/src/test/java/nl/haploid/event/channel/repository/RowChangeEventRepositoryIT.java
package nl.haploid.event.channel.repository;

import nl.haploid.event.channel.AbstractIT;
import nl.haploid.event.channel.TestData;
import org.junit.Test;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.transaction.annotation.Transactional;

import java.util.List;

import static org.junit.Assert.assertEquals;

public class RowChangeEventRepositoryIT extends AbstractIT {

	@Autowired
	private RowChangeEventDmoRepository repository;

	@Test
	@Transactional
	public void testFindAll() {
		final RowChangeEventDmo event = TestData.rowChangeEventDmo();
		repository.saveAndFlush(event);
		final List<RowChangeEventDmo> events = repository.findAll();
		assertEquals(1, events.size());
	}
}
